### DISCUSSION TITLE
Is it possible to pass a value from cli to wezterm (or read an env var) which can then be used within the config?

### DISCUSSION THREAD

**Discussion by MostHated** [wezterm/wezterm#2429] [General]
Hey all,
I was wondering if there was a way to either pass a value from cli to wezterm which can then be used within the config? The other option would be setting an env var when launching, if the config somehow has ability to read it? The main reason for this was simply to avoid having multiple config files to manage.

An example of what I was thinking would be to possibly have a predefined var in my config:

```lua
local my_var = ""

if my_var == "value_a" then
    do_something
elseif my_var == "value_b" then
    something_else
end
```

Then I would be able to run:

```sh
wezterm --my_var="value_a"
# or
wezterm --my_var="value_b"
```

Is there a way to accomplish this by chance? If not, is it possible to have something like:

```lua
local my_var = ENV_VAR
```
Then I could do something like this:
```sh
env ENV_VAR="value_a" wezterm
```

Thanks,
-MH

**Comment by wez** [comment]
You can use [os.getenv](https://www.lua.org/manual/5.4/manual.html#pdf-os.getenv) to read environment variables in lua.

You may also want to look at [user vars](https://wezfurlong.org/wezterm/config/lua/pane/get_user_vars.html) as a way to pass information from a process that was spawned inside wezterm to your lua config.

**Comment by MostHated** [comment]
It looks like the os.getenv should do the trick in this particular instance. The user vars seem quite interesting, though. I will have to look more into it.  Thanks for the info. :+1:
